src/lib/mongoBackend/mongoQueryContext.cpp

```diff
@@ -38,6 +38,268 @@
 
 /* ****************************************************************************
 *
+* someContextElementNotFound -
+*
+* Returns true if some attribut with 'found' set to 'false' is found in the CER vector passed
+* as argument
+*
+*/
+bool someContextElementNotFound(ContextElementResponseVector& cerV)
+{
+  for (unsigned int ix = 0; ix < cerV.size(); ++ix)
+  {
+    for (unsigned int jx = 0; jx < cerV.get(ix)->contextElement.contextAttributeVector.size(); ++jx)
+    {
+      if (!cerV.get(ix)->contextElement.contextAttributeVector.get(jx)->found)
+      {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+
+/* ****************************************************************************
+*
+* searchCprForAttribute -
+*
+* Search the CPr, given the entity/attribute as argument. Actually, two CPrs can be returned
```

Comment:
Yes, valid according to the compiler but not according the style guide.
We talked about thissome time ago and even looked up 'googles opinion' or something like that.

If a function is to change the value of a parameter, it should be visible from the outside, like this:

    int x = 19;
    changerFunction(&x);

That ampersand shows clearly that changerFunction might change the value of x.

Doing it the C++ reference way (horrible idea in my opinion), the fact that changerFunction might change the value of x is hidden.
This is very confusing, Very C++ ...

Now, what we decided was to send as real C-pointers all parameters that might be altered by the function.
If not altered, and 'complex' (like many strings), we use "const XXX& par", to send only the reference instead of copying the entire object on the stack.